Sources/Rendering/Core/ImageProperty/index.js

```diff
@@ -60,8 +60,14 @@ function vtkImageProperty(publicAPI, model) {
 
   // Set the piecewise function
   publicAPI.setPiecewiseFunction = (index, func) => {
```

Comment:
Why don't we use `(func, index=0)` if it is only new code that use the second argument?

Reply:
The reason this approach was taken previously was to maintain a consistent API for these functions throughout vtk.js / VTK -- the index comes first for functions like these.

Reply:
Yes, that was my inclination as well.  But the volume property uses the (index, func) ordering for similar methods, and supposedly other places as well, according to @thewtex, so he preferred it this way.

Reply:
Well we already decided to not follow VTK/C++ for `setInputData(ds, port=0)` as 95% of the case you only have 1 port. So 95% of the case, you can skip the port and have an easy to read code.

This use case is exactly the same... 

Since JS does not allow to easily override methods based on argument types, using C++ as reference is probably the best move.

Reply:
I did not know about such rule, but I guess it makes sense.
@jourdain , is there some documentation somewhere related to differences between VTK and VTK.JS ? Thanks.

Reply:
This is not really a rule but more something to take into account when you create a JavaScript library meant to be used by JS developers and not VTK/C++ developers. It is true that we want to map as much as possible to what has been done in VTK/C++ that worked well. But if we can fix design issue or provide improvements when possible, we should not prevent it just because *we don't follow the C++ way*.
We should enforce code simplicity and prevent user/developer error. VTK is complex enough.

I've been wanted to write more about the macro stuff and provide a real API doc and compatibility map, but unfortunately I never got a chance to do it. So I do not have anything more than [that](https://kitware.github.io/vtk-js/docs/develop_class.html) to offer.

@floryst did a great job explaining the new Widget architecture that is nowhere close to what VTK/C++ is doing but actually solve many issue that medical are having with VTK/C++.

Reply:
@jourdain yes, you are right -- it make sense to make API improvements when possible and that are appropriate to JavaScript. 

But, do you agree that there will be less errors when it is:

```
volumeProp.setRGBTransferFunction(index1, func1)
volumeProp.setScalarOpacity(index2, func2)
imageProp.setRGBTransferFunction(index3, func3)
imageProp.setPiecewiseFunction(index3, func4)
```

versus:

```
volumeProp.setRGBTransferFunction(index1, func1)
volumeProp.setScalarOpacity(index2, func2)
imageProp.setRGBTransferFunction(index3, func3)
imageProp.setPiecewiseFunction(func4, index4)
```

In the first, a new developer can confidently guess what the order is without referencing the API everytime they make a call or without knowing whether the history of whether it came from C++.